### PR TITLE
Ensure that headers/trailers/metadata are available after StopIteration

### PR DESCRIPTION
**Pull request by jplevyak**
until "continue" is called.  Includes test.

Signed-off-by: John Plevyak <jplevyak@gmail.com>

### CODE REVIEW

**Comment by bianpengyuan** [comment]
Fix https://github.com/envoyproxy/envoy-wasm/issues/458

File: source/extensions/common/wasm/context.cc [hunk 1]
```diff
@@ -1660,33 +1664,59 @@ void Context::onDelete() {
   }
 }
 
+void Context::continueRequest() {
+  if (decoder_callbacks_)
```
Comment (PiotrSikora) [inline source/extensions/common/wasm/context.cc]: Nit: missing brackets.

File: source/extensions/common/wasm/context.cc [hunk 2]
```diff
@@ -1660,33 +1664,59 @@ void Context::onDelete() {
...
+    decoder_callbacks_->continueDecoding();
+  request_headers_ = nullptr;
+  request_body_buffer_ = nullptr;
+  request_trailers_ = nullptr;
+  request_metadata_ = nullptr;
+}
+
+void Context::continueResponse() {
+  if (encoder_callbacks_)
```
Comment (PiotrSikora) [inline source/extensions/common/wasm/context.cc]: (same here)

File: source/extensions/common/wasm/context.cc (hunk 1 above)
Comment (jplevyak) [reply source/extensions/common/wasm/context.cc]: Done.

File: source/extensions/common/wasm/context.cc (hunk 2 above)
Comment (jplevyak) [reply source/extensions/common/wasm/context.cc]: Done

**Pull request merged**